test/functional/wallet_transactiontime_rescan.py

```diff
@@ -0,0 +1,160 @@
+        self.log.info('Send 1 btc to user')
+        miner_wallet.sendtoaddress(wo3, 1)
+
+        # generate more blocks and check blockcount
+        minernode.generatetoaddress(100, m1)
+        assert_equal(minernode.getblockcount(), 800)
+
+        # check user final balance and transaction count
+        self.log.info('Check user final balance')
+        assert_equal(wo_wallet.getbalance(), 16)
+        assert_equal(len(wo_wallet.listtransactions()), 3)
+
+        # check transactions time
+        self.log.info('Check transactions time')
+        for tx in wo_wallet.listtransactions():
+            if tx['address'] == 'bcrt1qqvs4h9g5kemwu8zdffv3c9tm3dm2qp9uzlxwzx':
+                assert_equal(tx['blocktime'], 1525971225)
+                assert_equal(tx['time'], 1525971225)
+            elif tx['address'] == 'bcrt1qnw7peeq89hwrg674zjf4phzzluld798p667q7q':
+                assert_equal(tx['blocktime'], 1526835225)
+                assert_equal(tx['time'], 1526835225)
+            elif tx['address'] == 'bcrt1qdr36v724uddf50jhnju3amzap2gmmt9tf4aac5':
+                assert_equal(tx['blocktime'], 1527699225)
+                assert_equal(tx['time'], 1527699225)
+
+        # restore user wallet without rescan
+        self.log.info('Restore user wallet on another node without rescan')
+        restorenode.createwallet(wallet_name='wo', disable_private_keys=True, )
+        restorewo_wallet = restorenode.get_wallet_rpc('wo')
+
+        restorewo_wallet.importpubkey(pubkey='02f8802962c2ca756d8961c719f75bdfff2b6dbb775788ed54a68bcb182c0ea1ef', rescan=False)
+        restorewo_wallet.importpubkey(pubkey='0330f4a96800a38fa0a49cde4b5f802e95a0c1044e5954389a8d00cecde01b842b', rescan=False)
+        restorewo_wallet.importpubkey(pubkey='02f5ccbee49546e0b5bf2e76422ed959e983dafaa94c8abafc9c8f8b99ef23d16a', rescan=False)
+
+        # check user has 0 balance and no transaction
+        assert_equal(restorewo_wallet.getbalance(), 0)
+        assert_equal(len(restorewo_wallet.listtransactions()), 0)
+
+        # proceed to rescan, first with an incomplete one, then with a full rescan
+        self.log.info('Rescan last history part')
+        restorewo_wallet.rescanblockchain(650)
+        self.log.info('Rescan all history')
+        restorewo_wallet.rescanblockchain()
+
+        # check user final balance and transaction count
+        self.log.info('Check user final balance after restoration')
+        assert_equal(restorewo_wallet.getbalance(), 16)
+        assert_equal(len(restorewo_wallet.listtransactions()), 3)
+
+        # check transactions time
+        self.log.info('Check transactions time after restoration')
+        for tx in restorewo_wallet.listtransactions():
+            if tx['address'] == 'bcrt1qqvs4h9g5kemwu8zdffv3c9tm3dm2qp9uzlxwzx':
+                assert_equal(tx['blocktime'], 1525971225)
+                assert_equal(tx['time'], 1525971225)
+            elif tx['address'] == 'bcrt1qnw7peeq89hwrg674zjf4phzzluld798p667q7q':
```

Comment:
The test does indeed fail here without the change.
```
2020-12-07T22:04:28.394000Z TestFramework (ERROR): Assertion failed
Traceback (most recent call last):
  File "/home/jon/projects/bitcoin/bitcoin/test/functional/test_framework/test_framework.py", line 126, in main
    self.run_test()
  File "test/functional/wallet_transactiontime_rescan.py", line 152, in run_test
    assert_equal(tx['time'], 1525971225)
  File "/home/jon/projects/bitcoin/bitcoin/test/functional/test_framework/util.py", line 50, in assert_equal
    raise AssertionError("not(%s)" % " == ".join(str(arg) for arg in (thing1, thing2) + args))
AssertionError: not(1527699225 == 1525971225)
```
But the test commit needs to be after the commit that makes the change, not before, otherwise the test fails. The tests and CI should pass for each commit. Also, you can squash the linter fixup commit into the test commit.

Reply:
done locally, not pushed yet